### PR TITLE
Migrated contributors(.js) to typescript(.tsx)

### LINKED ISSUE #97: pages/contributors.js

### PR DESCRIPTION
**Pull request by tiger-yash**
**Fixes**

Fixes #97 by @subodhk01 

**Description**

**Checklist:**

-   [x] I have read the contribution guidelines in the repository.
-   [x] My branch is up-to-date with upstream branch.
-   [x] My pull request has a descriptive title (not a vague title like "Update package.json").
-   [x] My code follows the established code style of the repository.
-   [x] I have tried running the project locally and verified that there are no visible errors.

### CODE REVIEW

**Comment by tiger-yash** [comment]
@subodhk01 @shubhanshu02 Please review

File: pages/contributors.tsx [hunk 1]
```diff
@@ -7,41 +7,64 @@ import { Spinner } from "react-bootstrap"
...
 
-const defaultOptions = {
+type ContributorSerializer = {
+    handle_name:string
+    name:string
+    handle_url:string
+    image:string
+    github:string
+    description:string[]
+}
```
Comment (shubhanshu02) [inline pages/contributors.tsx:17]: Rename this to something else. Serializers are used in the hackalog backend from Django Rest Framework (see [here](https://www.django-rest-framework.org/api-guide/serializers/)).

Simply, `Contributor` will work fine.

Comment (shubhanshu02) [reply pages/contributors.tsx:17]: Also mention in a comment above that objects of this type are received from the GitHub API.

File: pages/contributors.tsx [hunk 2]
```diff
@@ -7,41 +7,64 @@ import { Spinner } from "react-bootstrap"
...
+
+type ContriResponseSerializer = {
```
Comment (shubhanshu02) [inline pages/contributors.tsx:19]: If I am not wrong, is this the type for the Card component props?

If yes, we can reuse the same type from PR #108. Maybe it's better if we store the types in the `types` folder and import them wherever required. What do you think? @tiger-yash @Prakhar-Agarwal-byte

File: pages/contributors.tsx [hunk 3]
```diff
@@ -7,41 +7,64 @@ import { Spinner } from "react-bootstrap"
...
+    login:string
+    url:string
+    avatar_url:string
+    html_url:string
+}
+
+interface DefOptType {
```
Comment (shubhanshu02) [inline pages/contributors.tsx:26]: Maybe `DefaultOptionType` is more comprehensible?

File: pages/contributors.tsx [hunk 4]
```diff
@@ -7,41 +7,64 @@ import { Spinner } from "react-bootstrap"
...
+    loop: boolean;
+    autoplay: boolean;
+    animationData:any;
+  }
+
+const defaultOptions: DefOptType = {
```
Comment (shubhanshu02) [inline pages/contributors.tsx:32]: Just by looking at the name, we can't understand what this type is for. I guess we can consider renaming this to `defaultLottieOptions`. And similar for its type.

File: pages/contributors.tsx [hunk 5]
```diff
@@ -7,41 +7,64 @@ import { Spinner } from "react-bootstrap"
...
 
 export default function Contributors() {
-    const [contriFrontend, setContributorsFront] = useState()
-    const [contriBackend, setContributorsBack] = useState()
-    const [fullname, setFullName] = useState([{}])
-    const [contributors, setContributors] = useState()
-    const [error, setError] = useState(false)
+    const [contriFrontend, setContributorsFront] = useState<ContriResponseSerializer[]>(null)
+    const [contriBackend, setContributorsBack] = useState<ContriResponseSerializer[]>(null)
+    const [fullname, setFullName] = useState<Object[]>([{}] as Object[])
```
Comment (shubhanshu02) [inline pages/contributors.tsx:41]: Not sure about this part, but can we make it more specific than saying it to be an object? Like it'd be better if we know the properties of data objects in this array.

**Review by shubhanshu02 — CHANGES_REQUESTED** [review]
I left a few comments on the PR. Please have a look at them. 

Also, format the code using prettier (`npm run format`/`yarn format`) and merge the latest changes from this repo. Prefer rebasing in this case. (`git pull upstream main --rebase`)

File: pages/contributors.tsx (hunk 5 above)
Comment (tiger-yash) [reply pages/contributors.tsx:41]: problem is that the keys are not already known

File: pages/contributors.tsx (hunk 4 above)
Comment (tiger-yash) [reply pages/contributors.tsx:32]: fixed it

File: pages/contributors.tsx (hunk 3 above)
Comment (tiger-yash) [reply pages/contributors.tsx:26]: fixed it

File: pages/contributors.tsx (hunk 2 above)
Comment (tiger-yash) [reply pages/contributors.tsx:19]: `ContributorSerializer` is kind of common in both the files, but it is used only in these 2 files as it is basically produced specifically for contributors page using responses from github APIs.
It highly unlikely that we use it again, so we may also leave them be. I am open to other suggestions, though.

File: pages/contributors.tsx (hunk 1 above)
Comment (tiger-yash) [reply pages/contributors.tsx:17]: Done

**Comment by tiger-yash** [comment]
@shubhanshu02 please review

File: pages/contributors.tsx (hunk 5 above)
Comment (shubhanshu02) [reply pages/contributors.tsx:46]: No problem then.

**Review by shubhanshu02 — APPROVED** [review]
LGTM. 

Just one thing I just noticed in the `useEffect` for `contributors`, there might be a never-ending loop if we fix the ESLint warning here, causing the browser to crash. But since it not related to this PR, we'll deal with it later on some other issue.

**Comment by shubhanshu02** [comment]
Can you pull the latest changes from this repo? Thanks

**Pull request merged**